addons/gdshell/commands/default_commands/ls.gd

```diff
@@ -0,0 +1,119 @@
+	node_dict["has_script"] = node.get_script() != null
+	node_dict["script_file"] = node.get_script().get_path() if node_dict["has_script"] else "none"
+	node_dict["type"] = node.get_class()
+	node_dict["is_instanced_scene"] = node.scene_file_path != ""
+	node_dict["scene_file_path"] = str(node.scene_file_path)
+	node_dict["scene_tree_path"] = str(node.get_path())
+	node_dict["parent"] = node.get_parent().name if node.get_parent() != null else "none"
+	node_dict["children"] = []
+	
+	var child_count := node.get_child_count()
+	
+	for i in child_count:
+		#output(get_tree_dict(node.get_child(i), tree_dict, counter))
+		node_dict["children"].append(get_tree_dict(node.get_child(i), root_node, false))
+
+	if created_an_instance and node.get_parent() == root_node and node.get_index() == root_node.get_child_count() - 1:
+		done_with_instance.emit(root_node)
+
+	return node_dict
+
+
+func output_tree_dict(tree_dict: Dictionary, parent:= "", prefix:= "", root_node:= "", start:= true, last:= false):
+	var new_prefix := "" if start else " ┖╴" if last else " ┠╴"
+	root_node = tree_dict["name"] if start else root_node
+	var postfix = " (" + tree_dict["type"] + ") "
+	if tree_dict["is_instanced_scene"] and tree_dict["name"] != root_node:
+		postfix += "[hint=" + tree_dict["scene_file_path"] + "] 🎬[/hint]"
+	if tree_dict["has_script"]:
+		postfix += "[hint=" + tree_dict["script_file"] + "] 📜[/hint]"
+	
+	# Couldn't figure out how to connect meta_clicked signal
+	#var meta: String = "[url=" + tree_dict["scene_tree_path"] + "]"
+	#var name_output: String = meta + tree_dict["name"] + "[/url]"
+	var name_output
+	if tree_dict["scene_tree_path"] != "":
+		name_output = "[hint=" + tree_dict["scene_tree_path"] + "]" + tree_dict["name"] + "[/hint]"
+	else:
+		name_output = tree_dict["name"]
+	output(prefix + new_prefix + name_output + postfix)
+	
+	var dict_len:= len(tree_dict["children"]) if !tree_dict["is_instanced_scene"] or start else 0
+	
+	for i in dict_len:
+		var item = tree_dict["children"][i]
+		if item is Dictionary:
+			var num_of_siblings:= dict_len
+			if item["parent"] != root_node:
+				new_prefix = "   " if last or start else " ┃ "
+			output_tree_dict(item, item["parent"], prefix + new_prefix, root_node, false, i == num_of_siblings - 1)
+
+
+func get_starting_node(argv: Array):
+	var path: String
+	var node: Node
+	if len(argv) == 1:
+		return get_tree().current_scene
+	else:
+		path = argv[1]
+
+	var is_scn_file = path.ends_with(".tscn") or path.ends_with(".scn")
+	var is_resource_file = path.begins_with("res://")
+	var is_absolute_path = path.begins_with("/root") or path.begins_with("root")
+	
+	if is_resource_file and is_scn_file:
+		if !FileAccess.file_exists(path):
+			output("[color=red]" + "Resource file does not exit" + "[/color]")
+			return {"error": ERR_FILE_NOT_FOUND, "error_string": "File not found"}
+		else:
+			node = load(path).instantiate()
+			created_an_instance = true
+	
+	elif is_absolute_path:
+		node = get_node("/" + path)
```

Comment:
I'm not sure, but isn't this the intended order?
```suggestion
		node = get_node(path + "/")
```

Reply:
Oh sorry, i forgot to remove that! that line used to be part of another check, where i added the extra / in front just in case the user doesn't start their path with a /, since Godot ignores extra slashes. You are right it is not currently needed there.

Reply:
Thanks for the clarification! The comment would be nice for anyone working with it in the future.